entity-framework/core/modeling/value-comparers.md

```diff
@@ -0,0 +1,147 @@
+---
+title: Value Comparers - EF Core
+author: ajcvickers
+ms.date: 03/17/2020
+uid: core/modeling/value-comparers
+---
+
+# Value Comparers
+
+> [!NOTE]  
+> This feature is new in EF Core 3.0.
+
+> [!TIP]  
+> The code in this document can be found on GitHub as a [runnable sample](https://github.com/dotnet/EntityFramework.Docs/tree/master/samples/core/Modeling/ValueConversions/).
+
+## Background
+
+EF Core needs to compare property values when:
+
+* Determining whether or not a property has been changed as part of [detecting changes for updates](../saving/basic.md)
+* Determining whether or not two key values are the same when resolving relationships 
+
+This is handled automatically for common primitive types such as int, bool, DateTime, etc.
+
+For more complex types, choices need to be made as to how to do the comparison.
+For example, a byte array could be compared:
+
+* By reference, such that a difference is only detected if a new byte array is used
+* By deep comparison, such that mutation of the bytes in the array is detected
+
+EF Core uses the first of these approaches for non-key byte arrays.
+That is, only references are compared and a change is detected only when an existing byte array is replaced with a new one.
+This is a pragmatic decision that avoids deep comparison of many large byte arrays when executing SaveChanges.
+But the common scenario of replacing, say, an image with a different image is handled in a performant way.
+
+On the other hand, reference equality would not work when byte arrays are used to represent binary keys.
+It's very unlikely that an FK property is set to the _same instance_ as a PK property to which it needs to be compared.
+Therefore, EF Core uses deep comparisons for byte arrays acting as keys.
+This is unlikely to have a big performance hit since binary keys are usually short.
```

Comment:
new line for each sentence?

Reply:
https://github.com/dotnet/EntityFramework.Docs/pull/1335#issuecomment-469444982 :trollface: 

It makes diffing easier.